### PR TITLE
JS Adagrams - Katrina - Edges

### PR DESCRIPTION
**Pull request by kaganon**
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? |  Thinking about looping through elements and how they're both similar and also distinct from Ruby (i.e, the forEach loop, for loop, for..in, for..of loops). 
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? | I used almost the same strategies as I used for getting info with Ruby. Mostly Google, stackoverflow, reading through documentation, looking back on notes, and watching tutorials (scrimba, etc). 
What was a challenge you were able to overcome on this assignment? | Just getting used to the syntax, keeping track of my curly brackets/semicolons, the scope of my functions, and overall getting used to the weird nuances of JS. 
What is your favorite thing about learning a new programming language? | I think that because of my knowledge in Ruby, it feels pretty neat that learning a new language doesn't feel as overwhelming as when I first started. I can recognize similar patterns, read the documentation/google for info, and apply those skills to the new language. 
What is your least favorite thing about learning a new programming language? | The syntax and mixing up syntax from other languages. And finding that another language doesn't have a method you can call on (like .shuffle in Ruby - why don't you have it JS?!)
Do you have any recommendations on how we could improve this project for the next cohort? | Not really. I think it was great having the Ruby solution to adagrams to use so that I can focus more on getting used to the syntax and coding in JS :)

### CODE REVIEW

**Comment by tildeee** [comment]
# JS Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | x
Small commits with meaningful commit messages | x
**Code Requirements** |
`drawLetters` method | x
Uses appropriate data structure to store the letter distribution | x
All tests for `drawLetters` pass | x
`usesAvailableLetters` method | x
All tests for `usesAvailableLetters` pass | x
`scoreWord` method | x
Uses appropriate data structure to store the letter scores | x
All tests for `scoreWord` pass | x
`highestScoreFrom` method | x
Appropriately handles edge cases for tie-breaking logic | x
All tests for `highestScoreFrom` pass | x
**Overall** |

Well done on this project, Katrina! Overall, I see really great practice at various kinds of syntax in JavaScript, and good experimentation. I have a few comments, but in general I think you did a really great job. Good work!

File: src/adagrams.js [hunk 1]
```diff
@@ -1,7 +1,145 @@
...
+      const addToHand = () => {
+        return playerHand.push(randomLetter)
+      };
+
+      const removeLetter = () => {
+        return letterPool.splice(randomIndex, 1)
+      };
+
+      addToHand();
+      removeLetter();
```
Comment (tildeee) [inline src/adagrams.js:48]: Nice! These small, helper, inner functions are so small that they might seem like overkill at this stage. That being said, this project is for practicing syntax, and I think this is a really great time to practice it-- well done!

File: src/adagrams.js [hunk 2]
```diff
@@ -1,7 +1,145 @@
...
+    }
+
+    return playerHand;
+  },
+
+  usesAvailableLetters(word, drawn) {
+    const wordArray = word.toUpperCase().split('');
+    const letterHashObj = new Object();
```
Comment (tildeee) [inline src/adagrams.js:56]: Another way of saying this would be `const letterHashObj = {};`

File: src/adagrams.js [hunk 3]
```diff
@@ -1,7 +1,145 @@
...
+    let output = true
+
+    drawn.forEach(function (letter) {
```
Comment (tildeee) [inline src/adagrams.js:59]: You could use an arrow function here ;-)

File: src/adagrams.js [hunk 4]
```diff
@@ -1,7 +1,145 @@
...
+      if (letterHashObj[letter]) {
+        letterHashObj[letter] += 1;
+      }
+      else {
+        letterHashObj[letter] = 1;
+      }
+    });
+
+
+    wordArray.forEach(function (letter) {
```
Comment (tildeee) [inline src/adagrams.js:69]: You could use an arrow function here ;-)

File: src/adagrams.js [hunk 5]
```diff
@@ -1,7 +1,145 @@
...
+          output = false;
+        }
+        else {
+          letterHashObj[letter] -= 1;
+        }
+      }
+      else {
+        output = false;
+      }
+  });
```
Comment (tildeee) [inline src/adagrams.js:81]: watch your indentation!
